source/Calamari.Common/Features/StructuredVariables/XmlFormatVariableReplacer.cs

```diff
@@ -180,16 +179,18 @@ IEnumerable<NamespaceDeclaration> GetDeclaredNamespaceUris(XmlNode node)
                 }
         }
 
-        bool IsValidXPath(string xPath, IXmlNamespaceResolver nsResolver)
+        XPath2Expression TryGetXPathFromVariableKey(string variableKey, IXmlNamespaceResolver nsResolver)
```

Comment:
I'm surprised it's letting you get away without return type being `XPath2Expression?` in this method.

Reply:
@slewis74 I've been wondering about that. It currently doesn't let us use `?` on reference types. Have nullable reference types been completely enabled yet for Calamari?